### ISSUE TITLE
How to implement transfer learning in unseen dataset .?

### ISSUE THREAD

**Issue by TapendraBaduwal** [Deci-AI/super-gradients#920]
Lets if any unseen data we add then i do not want to train whole data set by mixing old and new data....I want to train only unseen data and our final model must detect old label and new label after training on unseen label?

**Comment by NatanBagrov** [comment]
Hello, could you please rephrase your question?

**Comment by TapendraBaduwal** [comment]
@NatanBagrov Lets my 1st model train on A ,B,C class...In next time i want to train My pre-train 1st model by adding D class ...(Note: In second time training I do not want to train and mix all data...i just want to train new D class). Final model must detect A, B, C, D class.

**Comment by NatanBagrov** [comment]
I understand.

Please close the new issue you created, #922 .

To achieve that you will need to [modify the head](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/models/detection_models/yolo_nas/dfl_heads.py#L51) to include additional new classes (=kernels), in your example, it has original 3 **kernels**, then you need to change to 4 kernels. Also, I would freeze the entire backbone, neck and original part of the head, and fine-tune only the "new" kernels.

This is somehow similar to #892

**Comment by TapendraBaduwal** [comment]
@NatanBagrov Can you guide me How to modify the head to include additional new classes (=kernels) i found only  kernel_size=3 ...i have to change self.num_classes am i right ? is there any config yml file to change self.num_classes from yml file ?

Also how to freeze the entire backbone, neck and original part of the head can you guide me i am new here from where i have to freeze dynamically....shall we need to unfeeze also?

**Comment by NatanBagrov** [comment]
Hello, I will elaborate with general guidelines, and will leave the implementation to you.
* If you change number of classes, you will not be able to re-use the checkpoint, so you need to do the modification *after* you have loaded the pre-trained checkpoint with the original number of classes.
* I would create a wrapper, that will take the model (already loaded), and will expand `cls_preds` with *the extended* filters (=kernels), that is, use the filters of the original classes, but also add uninitialized ones for the new classes. Then it is fine to change `num_classes`.
* To freeze parts of the model, you can read Torch documentation on how to achieve that. Here's a [reference](https://discuss.pytorch.org/t/how-the-pytorch-freeze-network-in-some-layers-only-the-rest-of-the-training/7088) to a forum thread with some info.


Good luck!

**Comment by TapendraBaduwal** [comment]
@NatanBagrov after freeze the entire backbone, neck and original part of the head....how my  fine-tune  "new" kernels learns without  backbone, neck and original part of the head ? Shall i need to add some layer for that?
